test/src/unit-max_tile_size.cc

```diff
@@ -0,0 +1,1237 @@
+    schema.set_domain(domain).set_order({{TILEDB_ROW_MAJOR, TILEDB_ROW_MAJOR}});
+
+    // Add one attribute "a1", so each (i) cell can store
+    // a nchars characters on "attr_name".
+    schema.add_attribute(
+        tiledb::Attribute::create<std::array<char, nchars>>(ctx, attr_name));
+    // schema.add_attribute(Attribute::create<char>(ctx, "a1"));
+    //   schema.add_attribute(Attribute::create<float[2]>(ctx, "a2"));
+
+    // Create the (empty) array on disk.
+    tiledb::Array::create(array_name, schema);
+  };
+
+  template <const int nchars = 1>
+  void write_array_attr(const std::string& array_name, const char* attr_name) {
+    tiledb::Context ctx;
+
+    // Prepare some data for the array
+    std::vector<std::array<char, nchars>> a;
+    for (auto i = 0; i < 4; ++i) {
+      std::array<char, nchars> buf;
+      memset(&buf[0], 'a' + i, nchars);
+      a.emplace_back(buf);
+    }
+
+    // Open the array for writing and create the query.
+    tiledb::Array array(ctx, array_name, TILEDB_WRITE);
+    tiledb::Query query(ctx, array);
+    query.set_layout(TILEDB_ROW_MAJOR).set_data_buffer(attr_name, a);
+
+    // Perform the write and close the array.
+    query.submit();
+    array.close();
+  };
+
+  uint64_t get_fragments_extreme(const std::string& array_uri) {
+    tiledb::Context ctx;
+    //tiledb_fragment_tile_size_extremes_t tiledb_fragment_tile_size_extremes;
+    uint64_t max_in_memory_tile_size = 0;
+    CHECK(
+        tiledb_array_maximum_tile_size(
+            &*ctx.ptr(),
+            array_uri.c_str(),
+            &max_in_memory_tile_size, 
+            nullptr) == TILEDB_OK);
+
+    if (showing_data) {
+      std::cout << "maximum in memory tile size "
+                << max_in_memory_tile_size
+                << std::endl;
+    }
+    return max_in_memory_tile_size;
+  }
+
+  uint64_t cpp_get_fragments_extreme(const std::string& array_uri) {
+    tiledb::Context ctx;
+    tiledb::Array array(ctx, array_uri, TILEDB_READ);
+    // tiledb_fragment_tile_size_extremes_t tiledb_fragment_tile_size_extremes;
+    uint64_t max_in_memory_tile_size = 0;
+    array.get_max_in_memory_tile_size(&max_in_memory_tile_size);
+
+    if (showing_data) {
+      std::cout << "maximum in memory tile size " << max_in_memory_tile_size
+                << std::endl;
+    }
+    return max_in_memory_tile_size;
+  }
+
+  tiledb::Context ctx_;
+  tiledb::VFS vfs_;
+};
+
+TEST_CASE_METHOD(
+    CAPI_MaxTileSizeFx,
+    "C++ API: Max tile size, dense, fixed, with consolidation",
+    "[capi][cppapi][max-tile-size][dense][consolidate]") {
+  // Name of array.
+  std::string array_name("fragments_consolidation_array");
+
+  auto create_array = [&]() -> void {
+    // Create a TileDB context.
+    tiledb::Context ctx;
+
+    remove_temp_dir(array_name.c_str());
+
+    // The array will be 4x4 with dimensions "rows" and "cols", with domain
+    // [1,4] and space tiles 2x2
+    tiledb::Domain domain(ctx);
+    domain
```

Comment:
Similar yes, but different as each array is slightly different.  I think it's better to have the individualized items local to the specific test case using them.

There is one templatized version in the fixture class because templatized lambdas aren't supported.